src/components/Form.js

```diff
@@ -1,27 +1,68 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../redux/books/books';
 import './styles/form.css';
 
 const Form = () => {
+  const dispatch = useDispatch();
+
+  const bookStore = () => ({
+    id: '',
+    title: '',
+    author: '',
+    category: '',
+  });
+
+  const [bookSetup, setBookSetup] = useState('');
+  const { title, author, category } = bookSetup;
+
+  const changeHandler = (event) => {
+    setBookSetup((prevState) => ({
+      ...prevState,
+      id: uuidv4(),
```

Comment:
### {OPTIONAL}

- [ ] It will be great to set up the id immediately in your bookStore object. Something like `id: uuidv4()`. This is because if your console.log your `bookSetup` state, you will see that every time a user adds anything in your form fields a new ID is generated. This is not the perfect implementation

Reply:
I've fixed all the errors. Thanks for the feedback